ExpectNonEmptyPlan Missing Data Source Differences Between v2.0.1 and v2.0.3

### SDK version

```
Terraform 0.12.29
v2.0.1 / v2.0.3
```

### Relevant provider source code

```go
func TestAccDataSourceAwsVpc_ExpectNonEmptyPlan(t *testing.T) {
	resource.ParallelTest(t, resource.TestCase{
		PreCheck:  func() { testAccPreCheck(t) },
		Providers: testAccProviders,
		Steps: []resource.TestStep{
			{
				Config: `
resource "aws_vpc" "test" {
  cidr_block = "10.0.0.0/16"
}

data "aws_vpc" "test" {
  # This shows a plan difference in Terraform 0.12
  # and should be caught by ExpectNonEmptyPlan
  depends_on = [aws_vpc.test]

  id = aws_vpc.test.id
}
`,
				ExpectNonEmptyPlan: true,
			},
		},
	})
}
```

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    aws = "3.8.0"
  }
  required_version = "0.12.29"
}

provider "aws" {
  region = "us-east-2"
}

resource "aws_vpc" "test" {
  cidr_block = "10.0.0.0/16"
}

data "aws_vpc" "test" {
  # This shows a plan difference in Terraform 0.12
  # and should be caught by ExpectNonEmptyPlan
  depends_on = [aws_vpc.test]

  id = aws_vpc.test.id
}
```

### Debug Output

Please let me know if you need something.

### Expected Behavior

v2.0.1:

```
--- PASS: TestAccDataSourceAwsVpc_ExpectNonEmptyPlan (19.21s)
```

### Actual Behavior

v2.0.3:

```
=== CONT  TestAccDataSourceAwsVpc_ExpectNonEmptyPlan
    data_source_aws_vpc_test.go:13: Step 1/1 error: Expected a non-empty plan, but got an empty plan!
--- FAIL: TestAccDataSourceAwsVpc_ExpectNonEmptyPlan (20.39s)
```

Maybe this conditional is no longer relevant due to the changes in terraform-plugin-test/terraform-exec?

https://github.com/hashicorp/terraform-plugin-sdk/blob/35d7fcb6c861606651dee1584261138dd71fb61b/helper/resource/testing_new.go#L166-L170

With `depends_on`:

```console
$ terraform plan -out=tfplan-depends_on &&  terraform show -json tfplan-depends_on | jq '.resource_changes'
[
  {
    "address": "aws_vpc.test",
    "mode": "managed",
    "type": "aws_vpc",
    "name": "test",
    "provider_name": "aws",
    "change": {
      "actions": [
        "no-op"
      ],
      "before": {
...
      },
      "after": {
...
      },
      "after_unknown": {}
    }
  },
  {
    "address": "data.aws_vpc.test",
    "mode": "data",
    "type": "aws_vpc",
    "name": "test",
    "provider_name": "aws",
    "change": {
      "actions": [
        "read"
      ],
      "before": null,
      "after": {
...
      },
      "after_unknown": {
...
      }
    }
  }
]
```

Without `depends_on`:

```console
$ terraform plan -out=tfplan && terraform show -json tfplan | jq '.resource_changes'
[
  {
    "address": "aws_vpc.test",
    "mode": "managed",
    "type": "aws_vpc",
    "name": "test",
    "provider_name": "aws",
    "change": {
      "actions": [
        "no-op"
      ],
      "before": {
...
      },
      "after": {
...
      },
      "after_unknown": {}
    }
  }
]
```

### Steps to Reproduce

Via acceptance testing framework:

1. `TF_ACC=1 go test -count=1 -run=TestAccDataSourceAwsVpc_ExpectNonEmptyPlan -timeout=2h -v`

Externally:

1. `terraform init`
1. `terraform apply`
1. `terraform apply` (shows data source plan difference)

### References

- Potentially https://github.com/hashicorp/terraform-plugin-sdk/issues/583 although that acceptance test is a managed resource and not a data source

--- a/helper/resource/testing_new.go
+++ b/helper/resource/testing_new.go
@@ -163,12 +163,6 @@ func stateIsEmpty(state *terraform.State) bool {
 
 func planIsEmpty(plan *tfjson.Plan) bool {
 	for _, rc := range plan.ResourceChanges {
-		if rc.Mode == tfjson.DataResourceMode {
-			// Skip data sources as the current implementation ignores
-			// existing state and they are all re-read every time
-			continue
-		}
-
 		for _, a := range rc.Change.Actions {
 			if a != tfjson.ActionNoop {
 				return false

